syne_tune/experiments/plotting.py

```diff
@@ -0,0 +1,664 @@
+         :class:`SubplotParameters`
+    :param show_init_trials: See :class:`ShowTrialParameters`
+    """
+
+    metric: str = None
+    mode: str = None
+    title: str = None
+    xlabel: str = None
+    ylabel: str = None
+    xlim: Tuple[float, float] = None
+    ylim: Tuple[float, float] = None
+    metric_multiplier: float = None
+    tick_params: Dict[str, Any] = None
+    aggregate_mode: str = None
+    dpi: int = None
+    grid: bool = None
+    subplots: SubplotParameters = None
+    show_init_trials: ShowTrialParameters = None
+
+    def merge_defaults(self, default_params: "PlotParameters") -> "PlotParameters":
+        new_params = _impute_with_defaults(
+            original=self,
+            default=default_params,
+            names=[
+                "metric",
+                "mode",
+                "title",
+                "xlabel",
+                "ylabel",
+                "xlim",
+                "ylim",
+                "metric_multiplier",
+                "tick_params",
+                "aggregate_mode",
+                "dpi",
+                "grid",
+            ],
+        )
+        default_values = [
+            ("metric", None),
+            ("mode", "min"),
+            ("title", ""),
+            ("metric_multiplier", 1),
+            ("aggregate_mode", DEFAULT_AGGREGATE_MODE),
+            ("dpi", 200),
+            ("grid", False),
+            ("xlabel", DEFAULT_XLABEL),
+        ]
+        _check_and_set_defaults(new_params, default_values)
+        if self.subplots is None:
+            new_params["subplots"] = default_params.subplots
+        elif default_params.subplots is None:
+            new_params["subplots"] = self.subplots
+        else:
+            new_params["subplots"] = self.subplots.merge_defaults(
+                default_params.subplots
+            )
+        if self.show_init_trials is None:
+            new_params["show_init_trials"] = default_params.show_init_trials
+        elif default_params.show_init_trials is None:
+            new_params["show_init_trials"] = self.show_init_trials
+        else:
+            new_params["show_init_trials"] = self.show_init_trials.merge_defaults(
+                default_params.show_init_trials
+            )
+        return PlotParameters(**new_params)
+
+
+class ComparativeResults:
+    """
+    This class loads, processes, and plots results of a comparative study,
+    combining several experiments for different methods, seeds, and
+    benchmarks (optional). Note that an experiment corresponds to one run
+    of HPO, resulting in files :const:`~syne_tune.constants.ST_METADATA_FILENAME`
+    for metadata, and :const:`~syne_tune.constants.ST_RESULTS_DATAFRAME_FILENAME`
+    for time-stamped results.
+
+    There is one comparative plot per benchmark (aggregation of results
+    across benchmarks are not supported here). Results are grouped by
+    setup (which usually equates to method), and then summary statistics are
```

Comment:
`setup` might be confusing here. Maybe trajectory or just method?

Reply:
Method would not be good.